web/yo/app/scripts/controllers/gene.js

```diff
@@ -1871,6 +1883,16 @@ angular.module('oncokbApp')
                 } else if (type === 'treatment') {
                     obj = $scope.gene.mutations[indices[0]].tumors[indices[1]].TIs[indices[2]].treatments[indices[3]];
                     uuid = $scope.gene.mutations[indices[0]].tumors[indices[1]].TIs[indices[2]].treatments[indices[3]].name_uuid;
+                    //Map should be changed. More code here
+                    drugUuids = $scope.gene.mutations[indices[0]].tumors[indices[1]].TIs[indices[2]].treatments[indices[3]].name;
+                    drugUuids = drugUuids.replace(" + ", " ");
+                    drugUuids = drugUuids.replace(", ", " ");
+                    var keys = new Array();
+                    if(drugUuids != undefined)
```

Comment:
It's like undefined check in translator. While adding therapy, it’s possible that saving information in Firebase hasn’t done while translator working. There may be a TypeError because of undefined data.

Reply:
If that's the case, isn't it supposed to check before line 1888?